### PR TITLE
[SYCLomatic][CMake] Add support for cmake cuda toolkit and sdk

### PR DESCRIPTION
**Pull request by the-slow-one**
Add support for 
* CUDA_TOOLKIT_ROOT_DIR
* CUDA_TOOLKIT_INCLUDE
* CUDAToolkit_LIBRARY_DIR

### CODE REVIEW

File: clang/tools/dpct/cmake/dpct.cmake [hunk 1]
```diff
@@ -119,4 +119,9 @@ set(CMAKE_SYCL_FLAGS "")
 set(COMPATIBILITY_VERSION 99.9)
 set(COMPATIBILITY_VERSION_MAJOR 99)
 set(COMPATIBILITY_VERSION_MINOR 9)
-set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
\ No newline at end of file
+set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
+set(SYCL_TOOLKIT_ROOT_DIR "${ONEAPI_ROOT}")
+set(SYCL_TOOLKIT_INCLUDE "${ONEAPI_ROOT}/vtune/latest/include;${ONEAPI_ROOT}/vtune/latest/sdk/include")
+# OneAPI library is marked for linker lookup by FindIntelSYCL.cmake. So we can
+# safely set the CUDAToolkit_LIBRARY_DIR to ""
```
Comment (tomflinda) [inline clang/tools/dpct/cmake/dpct.cmake]: 1. According to the definition of CUDAToolkit_LIBRARY_DIR in https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html:
```
The path to the CUDA Toolkit library directory that contains the CUDA Runtime library cudart.
```
While in the SYCL side, the SYCL runtime library is auto-loaded when option "-fsycl" is specified, that is why we can set the helper variable SYCLToolkit_LIBRARY_DIR to ""
2.  There is no file "FindIntelSYCL.cmake" in oneAPI package, pls provide the correct file name you refer to.

Comment (the-slow-one) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
# The SYCL runtime library is auto-loaded when option `-fsycl` is specified. So we can
# safely set the CUDAToolkit_LIBRARY_DIR to ""
```

Comment (the-slow-one) [reply clang/tools/dpct/cmake/dpct.cmake]: Done

File: clang/tools/dpct/cmake/dpct.cmake [hunk 2]
```diff
@@ -119,4 +119,9 @@ set(CMAKE_SYCL_FLAGS "")
 set(COMPATIBILITY_VERSION 99.9)
 set(COMPATIBILITY_VERSION_MAJOR 99)
 set(COMPATIBILITY_VERSION_MINOR 9)
-set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
\ No newline at end of file
+set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
+set(SYCL_TOOLKIT_ROOT_DIR "${ONEAPI_ROOT}")
+set(SYCL_TOOLKIT_INCLUDE "${ONEAPI_ROOT}/vtune/latest/include;${ONEAPI_ROOT}/vtune/latest/sdk/include")
```
Comment (tomflinda) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
set(SYCL_TOOLKIT_INCLUDE SYCL_INCLUDE_DIR)
```

File: clang/tools/dpct/cmake/dpct.cmake [hunk 3]
```diff
@@ -119,4 +119,9 @@ set(CMAKE_SYCL_FLAGS "")
...
+# The SYCL runtime library is auto-loaded when option `-fsycl` is specified. So we can
+# safely set the CUDAToolkit_LIBRARY_DIR to ""
+set(SYCLToolkit_LIBRARY_DIR "")
```
Comment (tomflinda) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
set(SYCLToolkit_LIBRARY_DIR ${SYCL_INCLUDE_DIR}/../lib)
```

File: clang/tools/dpct/cmake/dpct.cmake (hunk 2 above)
Comment (the-slow-one) [reply clang/tools/dpct/cmake/dpct.cmake]: Done

Comment (the-slow-one) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
set(SYCL_TOOLKIT_INCLUDE "${SYCL_INCLUDE_DIR}")
```

File: clang/tools/dpct/cmake/dpct.cmake (hunk 3 above)
Comment (the-slow-one) [reply clang/tools/dpct/cmake/dpct.cmake]: Done

Comment (the-slow-one) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
set(SYCLToolkit_LIBRARY_DIR "${SYCL_INCLUDE_DIR}/../lib")
```

File: clang/tools/dpct/cmake/dpct.cmake [hunk 4]
```diff
@@ -119,4 +119,9 @@ set(CMAKE_SYCL_FLAGS "")
...
 set(COMPATIBILITY_VERSION_MAJOR 99)
 set(COMPATIBILITY_VERSION_MINOR 9)
-set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
\ No newline at end of file
+set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
+set(SYCL_TOOLKIT_ROOT_DIR "${ONEAPI_ROOT}")
+set(SYCL_TOOLKIT_INCLUDE "${SYCL_INCLUDE_DIR}")
+# The SYCL runtime library is auto-loaded when option `-fsycl` is specified. So we can
+# safely set the CUDAToolkit_LIBRARY_DIR to ""
+set(SYCLToolkit_LIBRARY_DIR "${SYCL_INCLUDE_DIR}/../lib")
```
Comment (the-slow-one) [inline clang/tools/dpct/cmake/dpct.cmake:126]:
```suggestion
set(SYCLToolkit_LIBRARY_DIR "${SYCL_LIBRARY_DIR}")
```

File: clang/tools/dpct/cmake/dpct.cmake [hunk 5]
```diff
@@ -119,4 +119,9 @@ set(CMAKE_SYCL_FLAGS "")
...
 set(COMPATIBILITY_VERSION_MAJOR 99)
 set(COMPATIBILITY_VERSION_MINOR 9)
-set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
\ No newline at end of file
+set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
+set(SYCL_TOOLKIT_ROOT_DIR "${ONEAPI_ROOT}")
+set(SYCL_TOOLKIT_INCLUDE "${SYCL_INCLUDE_DIR}")
+# The SYCL runtime library is auto-loaded when option `-fsycl` is specified. So we can
+# safely set the CUDAToolkit_LIBRARY_DIR to ""
+set(SYCLToolkit_LIBRARY_DIR "${SYCL_LIBRARY_DIR}")
```
Comment (the-slow-one) [inline clang/tools/dpct/cmake/dpct.cmake]:
```suggestion
set(SYCLToolkit_LIBRARY_DIR "${SYCL_INCLUDE_DIR}/../lib")
```

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**